src/boot/bootctl.c

```diff
@@ -664,9 +669,17 @@ static int install_binaries(const char *esp_path, bool force) {
         FOREACH_DIRENT(de, d, return log_error_errno(errno, "Failed to read \""BOOTLIBDIR"\": %m")) {
                 int k;
 
-                if (!endswith_no_case(de->d_name, ".efi"))
+                if (!endswith_no_case(de->d_name, ".efi") && !endswith_no_case(de->d_name, ".efi.signed"))
                         continue;
 
+                /* skip the .efi file, if there's a .signed version of it */
+                if (endswith_no_case(de->d_name, ".efi")) {
+                        const char *s;
+                        s = strjoina(BOOTLIBDIR, "/", de->d_name, ".signed");
```

Comment:
Oops, lgtm is right: strjoina cannot be used here. Please use normal strjoin instead.

Reply:
right. 
if I add the `_cleanup_free_` attribute to `const char *s` will it be freed on each loop?

Reply:
> if I add the _cleanup_free_ attribute to const char *s will it be freed on each loop?

Right.

Reply:
fixed